packages/block-editor/src/components/inserter/category-tabs/index.js

```diff
@@ -35,14 +35,13 @@ function CategoryTabs( {
 	const selectedTabId = selectedCategory ? selectedCategory.name : null;
 	const [ activeTabId, setActiveId ] = useState();
 	const firstTabId = categories?.[ 0 ]?.name;
-	useEffect( () => {
-		// If there is no active tab, make the first tab the active tab, so that
-		// when focus is moved to the tablist, the first tab will be focused
-		// despite not being selected
-		if ( selectedTabId === null && ! activeTabId && firstTabId ) {
-			setActiveId( firstTabId );
-		}
-	}, [ selectedTabId, activeTabId, firstTabId, setActiveId ] );
+
+	// If there is no active tab, make the first tab the active tab, so that
+	// when focus is moved to the tablist, the first tab will be focused
+	// despite not being selected
+	if ( selectedTabId === null && ! activeTabId && firstTabId ) {
+		setActiveId( firstTabId );
```

Comment:
We shouldn't set state during render?

Reply:
@ellatrix see https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes for the pattern followed here.

Reply:
You can if it's called conditionally; it prevents extra render.

Refs:

* https://react.dev/reference/react/useState#storing-information-from-previous-renders
* https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes
* https://blog.isquaredsoftware.com/2020/05/blogged-answers-a-mostly-complete-guide-to-react-rendering-behavior/#setting-state-while-rendering